### PR TITLE
Make commands.log support rotate mechanism and fix bug#160-165

### PR DESCRIPTION
**Pull request by hu-weihua**
Fix bugs:
#160 The /var/log/xcat/commands.log needs a better format of datetime 
#162 Security issue of the xcatd command log writer
#163 The xcatd command log writer failed to get the error message of nodels
#164 offline trouble shooting :could not cover rcons getcons text error info in commands.log
#165 The xcatd command log writer needs a better method to handle command line arguments

New feature
Make commands.log support rotate mechanism

### CODE REVIEW

**Comment by daniceexi** [comment]
I have reviewed the change and agree to check in.

File: xCAT-server/sbin/xcatd [hunk 1]
```diff
@@ -2450,7 +2502,12 @@ sub service_connection {
     
     if(exists($req->{arg})){
         foreach my $arg (@{$req->{arg}}) {
-	        $cmdlog_alllog .= $arg." ";
+            if($arg =~ /[^A-Za-z0-9.-]/){
+                $arg =~ s/'/'\\''/g;
```
Comment (immarvin) [inline xCAT-server/sbin/xcatd]: hi, this line might caused the issue https://github.com/xcat2/xcat-core/issues/203

**Comment by daniceexi** [comment]
@immarvin Could help to check whether it could fix the issue if removing the change on $arg.

**Comment by immarvin** [comment]
yes.

When this line is commented out, the request passed to xdsh.pm by ``xdsh c910f03c01p08 "echo 'test' >/tmp/test"`` will be:
````
...
            'arg' => [
                       'echo \'test\' >/tmp/test'
                     ],
...
````

**Pull request merged**